ngraph/core/include/ngraph/op/matrix_nms.hpp

```diff
@@ -0,0 +1,138 @@
+                          const ngraph::element::Type& output_type = ngraph::element::i64,
+                          const float score_threshold = 0.0f,
+                          const int nms_top_k = -1,
+                          const int keep_top_k = -1,
+                          const int background_class = -1,
+                          const DecayFunction decay_function = DecayFunction::LINEAR,
+                          const float gaussian_sigma = 2.0f,
+                          const float post_threshold = 0.0f,
+                          const bool normalized = true);
+
+                bool visit_attributes(AttributeVisitor& visitor) override;
+
+                std::shared_ptr<Node>
+                    clone_with_new_inputs(const OutputVector& new_args) const override;
+
+                bool get_sort_result_across_batch() const { return m_sort_result_across_batch; }
+                void set_sort_result_across_batch(const bool sort_result_across_batch)
+                {
+                    m_sort_result_across_batch = sort_result_across_batch;
+                }
+
+                float get_score_threshold() const { return m_score_threshold; }
+                void set_score_threshold(const float score_threshold)
+                {
+                    m_score_threshold = score_threshold;
+                }
+
+                int get_background_class() const { return m_background_class; }
+                void set_background_class(const int background_class)
+                {
+                    m_background_class = background_class;
+                }
+
+                DecayFunction get_decay_function() const { return m_decay_function; }
+                void set_decay_function(const DecayFunction decay_function)
+                {
+                    m_decay_function = decay_function;
+                }
+
+                float get_gaussian_sigma() const { return m_gaussian_sigma; }
+                void set_gaussian_sigma(const float gaussian_sigma)
+                {
+                    m_gaussian_sigma = gaussian_sigma;
+                }
+
+                float get_post_threshold() const { return m_post_threshold; }
+                void set_post_threshold(const float post_threshold)
+                {
+                    m_post_threshold = post_threshold;
+                }
+
+                bool get_normalized() const { return m_normalized; }
+                void set_normalized(const bool normalized) { m_normalized = normalized; }
+
+            protected:
+                bool m_sort_result_across_batch;
+                float m_score_threshold;
+                int m_background_class;
+                DecayFunction m_decay_function;
+                float m_gaussian_sigma;
+                float m_post_threshold;
+                bool m_normalized;
+                void validate() override;
+            };
+        } // namespace v8
+    }     // namespace op
```

Comment:
Could you create a structure with all attributes of the operation instead of attributes as arguments of the constructor?
```suggestion
                enum class DecayFunction
                {
                    GAUSSIAN,
                    LINEAR
                };
                
                struct Attributes
                {
                          SortResultType sort_result_type = SortResultType::NONE;
                          bool sort_result_across_batch = false;
                          ngraph::element::Type& output_type = ngraph::element::i64;
                          float score_threshold = 0.0f;
                          int nms_top_k = -1;
                          int keep_top_k = -1;
                          int background_class = -1;
                          DecayFunction decay_function = DecayFunction::LINEAR;
                          float gaussian_sigma = 2.0f;
                          float post_threshold = 0.0f;
                          bool normalized = true;
                };

                MatrixNms() = default;

                /// \brief Constructs a MatrixNms operation
                ///
                /// \param boxes Node producing the box coordinates
                /// \param scores Node producing the box scores
                /// \param attrs Attributes of the operation
                MatrixNms(const Output<Node>& boxes,
                          const Output<Node>& scores,
                          const Attributes attrs);
                          
                bool visit_attributes(AttributeVisitor& visitor) override;

                std::shared_ptr<Node>
                    clone_with_new_inputs(const OutputVector& new_args) const override;

                Attributes get_attrs() const { return m_attrs; }
                
                void set_attrs(const Attributes& attrs) {m_attrs = attrs;}
            protected:
                Attributes m_attrs;
                
                void validate() override;
            };
        } // namespace v8
    }     // namespace op
```
The cpp-file should be correspondingly changed. Of course, we need comments before structure `Attributes`.

Reply:
Done